runtime/intern.c

```diff
@@ -239,446 +284,395 @@ static void readfloats(struct caml_intern_state* s,
+        header = (header_t) read64u(s);
+        tag = Tag_hd(header);
+        size = Wosize_hd(header);
+        goto read_block;
 #endif
-            v = caml_alloc_custom(ops, expected_size, 0, 0);
-            size = ops->deserialize(Data_custom_val(v));
-            if (size != expected_size) {
-              intern_cleanup(s);
-              caml_failwith(
-                "input_value: incorrect length of serialized custom block");
-            }
-            if (use_intern_table)
-              Store_field (intern_obj_table, obj_counter++, v);
-            break;
-          }
-          default:
+      case CODE_STRING8:
+        len = read8u(s);
+        goto read_string;
+      case CODE_STRING32:
+        len = read32u(s);
+        goto read_string;
+#ifdef ARCH_SIXTYFOUR
+      case CODE_STRING64:
+        len = read64u(s);
+        goto read_string;
+#endif
+      case CODE_DOUBLE_LITTLE:
+      case CODE_DOUBLE_BIG:
+        v = intern_alloc_obj (s, d, Double_wosize, Double_tag);
+        if (s->intern_obj_table != NULL)
+          s->intern_obj_table[s->obj_counter++] = v;
+        readfloat(s, (double *) v, code);
+        break;
+      case CODE_DOUBLE_ARRAY8_LITTLE:
+      case CODE_DOUBLE_ARRAY8_BIG:
+        len = read8u(s);
+      read_double_array:
+        size = len * Double_wosize;
+        v = intern_alloc_obj (s, d, size, Double_array_tag);
+        if (s->intern_obj_table != NULL)
+          s->intern_obj_table[s->obj_counter++] = v;
+        readfloats(s, (double *) v, len, code);
+        break;
+      case CODE_DOUBLE_ARRAY32_LITTLE:
+      case CODE_DOUBLE_ARRAY32_BIG:
+        len = read32u(s);
+        goto read_double_array;
+#ifdef ARCH_SIXTYFOUR
+      case CODE_DOUBLE_ARRAY64_LITTLE:
+      case CODE_DOUBLE_ARRAY64_BIG:
+        len = read64u(s);
+        goto read_double_array;
+#endif
+      case CODE_CODEPOINTER:
+        ofs = read32u(s);
+        readblock(s, digest, 16);
+        codeptr = intern_resolve_code_pointer(digest, ofs);
+        if (codeptr != NULL) {
+          v = (value) codeptr;
+        } else {
+          const value * function_placeholder =
+            caml_named_value ("Debugger.function_placeholder");
+          if (function_placeholder != NULL) {
+            /* Use the code pointer from the "placeholder" function */
+            v = (value) Code_val(*function_placeholder);
+          } else {
             intern_cleanup(s);
-            caml_failwith("input_value: ill-formed message");
+            intern_bad_code_pointer(digest);
           }
         }
+        break;
+      case CODE_INFIXPOINTER:
+        ofs = read32u(s);
+        /* Read a value to *dest, then offset *dest by ofs */
+        PushItem(s);
+        sp->dest = dest;
+        sp->op = OShift;
+        sp->arg = ofs;
+        ReadItems(s, dest, 1);
+        continue;  /* with next iteration of main loop, skipping *dest = v */
+      case CODE_CUSTOM:
+      case CODE_CUSTOM_LEN:
+      case CODE_CUSTOM_FIXED: {
```

Comment:
In trunk there's code marked "deprecated" that handles the `CODE_CUSTOM` case specially.  I don't see this code here.  Either we handle `CODE_CUSTOM` or we don't...

Reply:
Removed `CODE_CUSTOM` which we don't handle anymore.